src/main/java/org/apache/mesos/scheduler/plan/DefaultPlanScheduler.java

```diff
@@ -0,0 +1,113 @@
+package org.apache.mesos.scheduler.plan;
+
+import com.google.inject.Inject;
+import org.apache.mesos.Protos;
+import org.apache.mesos.SchedulerDriver;
+import org.apache.mesos.offer.*;
+import org.apache.mesos.scheduler.TaskKiller;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Default deployment scheduler. See docs in {@link PlanScheduler} interface.
+ */
+public class DefaultPlanScheduler implements PlanScheduler {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultPlanScheduler.class);
+
+    private final OfferAccepter offerAccepter;
+    private final OfferEvaluator offerEvaluator;
+    private final TaskKiller taskKiller;
+
+    @Inject
+    public DefaultPlanScheduler(OfferAccepter offerAccepter, TaskKiller taskKiller) {
+        this(offerAccepter, new OfferEvaluator(), taskKiller);
+    }
+
+    public DefaultPlanScheduler(OfferAccepter offerAccepter, OfferEvaluator offerEvaluator, TaskKiller taskKiller) {
+        this.offerAccepter = offerAccepter;
+        this.offerEvaluator = offerEvaluator;
+        this.taskKiller = taskKiller;
+    }
+
+    @Override
+    public List<Protos.OfferID> resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers, Block block) {
+        List<Protos.OfferID> acceptedOffers = new ArrayList<>();
+
+        if (driver == null || offers == null) {
+            logger.error("Unexpected null argument encountered: driver='{}' offers='{}'", driver, offers);
+            return acceptedOffers;
+        }
+
+        if (block == null) {
+            logger.info("Ignoring resource offers for null block.");
+            return acceptedOffers;
+        }
+
+        if (!block.isPending()) {
+            logger.info("Ignoring resource offers for block: {} status: {}", block.getName(), Block.getStatus(block));
+            return acceptedOffers;
+        }
+
+        logger.info("Processing resource offers for block: {}", block.getName());
+        Optional<OfferRequirement> offerRequirementOptional = block.start();
+        if (!offerRequirementOptional.isPresent()) {
+            logger.info("No OfferRequirement for block: {}", block.getName());
+            block.updateOfferStatus(Optional.empty());
+            return acceptedOffers;
+        }
+
+        OfferRequirement offerRequirement = offerRequirementOptional.get();
+        // It is harmless to attempt to kill tasks which have never been launched.
```

Comment:
What does killTasks() accomplish? Delete data from StateStore? (maybe add a comment describing the answer)